euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/Collector.java

```diff
@@ -29,4 +29,13 @@
    * @param elem the element to collect
    */
   void collect(T elem);
+
+  /**
+   * Returns {@link Context} view of the collector.
+   * Since {@link Collector} usually share the same methods
+   * as {@link Context} it can be safely casted.
+   *
+   * @return this instance as a context class
+   */
+  Context asContext();
```

Comment:
This method seems that every implementation will have no other option than just to cast the `Collector` to `Context`, am I right? We could move that code inside either default implementation of this method, or maybe just cast it directly instead of calling this method.